### PR TITLE
feat: fix header bugs, improve UI, optimize touchscreen experience

### PR DESCRIPTION
**Pull request by turbodog03**
# changes:
* Fix header overflow 
image
* Fix account dropdown disappear 
image
* Add background color and card to warp content
* Add scroll for tables
* Optimize tables for touchscreen, the table won't overflow anymore
image
image
* Improve the dropdown UI when hover

# Preview:
http://39.106.18.82/

### CODE REVIEW

**Comment by turbodog03** [comment]
Please review @nodece

File: web/src/App.js [hunk 1]
```diff
@@ -376,40 +382,101 @@ class App extends Component {
...
+          <div style = {{float: 'right'}}>
+          {
+            this.renderAccount()
+          }
+          <SelectLanguageBox/>
+        </div>
+        </Menu>
+      </Header>
+      <Switch>
+        <Route exact path="/result" render={(props) => this.renderHomeIfLoggedIn(<ResultPage {...props} />)}/>
```
Comment (hsluoyz) [inline web/src/App.js:456]: Why repeat the routes?

File: web/src/UserListPage.js [hunk 2]
```diff
@@ -14,11 +14,12 @@
 
 import React from "react";
 import {Link} from "react-router-dom";
-import {Button, Col, Popconfirm, Row, Switch, Table} from 'antd';
+import {Button, Card, Col, Popconfirm, Row, Switch, Table} from 'antd';
 import moment from "moment";
 import * as Setting from "./Setting";
 import * as UserBackend from "./backend/UserBackend";
 import i18next from "i18next";
+import { Content } from "antd/lib/layout/layout";
```
Comment (hsluoyz) [inline web/src/UserListPage.js:22]: I don't think we should import "antd/lib/layout/layout"

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: web/src/LdapTable.js [hunk 3]
```diff
@@ -18,6 +18,7 @@ import * as Setting from "./Setting";
 import i18next from "i18next";
 import * as LdapBackend from "./backend/LdapBackend";
 import {Link} from "react-router-dom";
+import { Content } from "antd/lib/layout/layout";
```
Comment (hsluoyz) [inline web/src/LdapTable.js:21]: Content is not used?

File: web/src/App.js [hunk 4]
```diff
@@ -376,40 +409,61 @@ class App extends Component {
...
-          <Route exact path="/applications/:applicationName" render={(props) => this.renderLoginIfNotLoggedIn(<ApplicationEditPage account={this.state.account} {...props} />)}/>
-          <Route exact path="/ldap/:ldapId" render={(props) => this.renderLoginIfNotLoggedIn(<LdapEditPage account={this.state.account} {...props} />)}/>
-          <Route exact path="/ldap/sync/:ldapId" render={(props) => this.renderLoginIfNotLoggedIn(<LdapSyncPage account={this.state.account} {...props} />)}/>
-          <Route exact path="/tokens" render={(props) => this.renderLoginIfNotLoggedIn(<TokenListPage account={this.state.account} {...props} />)}/>
-          <Route exact path="/tokens/:tokenName" render={(props) => this.renderLoginIfNotLoggedIn(<TokenEditPage account={this.state.account} {...props} />)}/>
-          <Route exact path="/records" render={(props) => this.renderLoginIfNotLoggedIn(<RecordListPage account={this.state.account} {...props} />)}/>
-        </Switch>
+        <Layout style={{backgroundColor: "#f5f5f5", alignItems: 'stretch'}}>
+          <Card className="content-warp-card">
+              {this.renderRouter()}
```
Comment (hsluoyz) [inline web/src/App.js:427]: {
xxx
}

File: web/src/App.js [hunk 5]
```diff
@@ -376,40 +409,61 @@ class App extends Component {
...
+          }
+          <div style = {{float: 'right'}}>
+          {
+            this.renderAccount()
+          }
+          <SelectLanguageBox/>
+        </div>
+        </Menu>
+      </Header>
+      {this.renderRouter()}
```
Comment (hsluoyz) [inline web/src/App.js:461]: {
xxx
}

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Comment by hsluoyz** [comment]
@Kininaru @sh1luo plz review.

**Review by Kininaru — APPROVED** [review]
Mobile UI still looks strange, and some display bugs: 

image

Maybe we can fix these bugs in future?

**Comment by hsluoyz** [comment]
@Kininaru OK, let's fix the mobile view in next PR.

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**